src/include/mpir_request.h

```diff
@@ -567,20 +567,6 @@ int MPIR_Grequest_cancel(MPIR_Request * request_ptr, int complete);
 int MPIR_Grequest_query(MPIR_Request * request_ptr);
 int MPIR_Grequest_free(MPIR_Request * request_ptr);
 
-int MPIR_Grequest_complete_impl(MPIR_Request * request_ptr);
-int MPIR_Grequest_start_impl(MPI_Grequest_query_function * query_fn,
-                             MPI_Grequest_free_function * free_fn,
-                             MPI_Grequest_cancel_function * cancel_fn,
-                             void *extra_state, MPIR_Request ** request_ptr);
-int MPIR_Grequest_class_create_impl(MPI_Grequest_query_function * query_fn,
-                                    MPI_Grequest_free_function * free_fn,
-                                    MPI_Grequest_cancel_function * cancel_fn,
-                                    MPIX_Grequest_poll_function * poll_fn,
-                                    MPIX_Grequest_wait_function * wait_fn,
-                                    MPIX_Grequest_class * greq_class);
-int MPIR_Grequest_class_allocate_impl(MPIX_Grequest_class greq_class,
-                                      void *extra_state, MPIR_Request ** p_request_ptr);
-
 /* These routines below are helpers for the Extended generalized requests. */
```

Comment:
They are not removed. They are renamed to use the consistent pattern of MPIR_Xxx_impl. MPIX_Grequest_start_impl is removed since the name doesn't fit into MPIR_Xxx_impl , thus can be confusing. Instead we will have MPIX_Grequest_start call MPIR_Grequest_start_impl and set the additional poll_fn and wait_fn as we mentioned in the other comment.